bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/handler/TadoHomeHandler.java

```diff
@@ -156,7 +168,24 @@ private synchronized void initializeBridgeStatusAndPropertiesIfOffline() {
                     return;
                 }
 
+                /*
+                 * If there is only one home, or if there is no valid configuration.homeId entry, then use the first
+                 * home id. Otherwise use the configuration.homeId entry value (if one exists). If there is no valid
+                 * configuration.homeId entry but there are multiple homes then log the available home to help the
+                 * user set up the proper configuration.homeId entry.
+                 */
                 Integer firstHomeId = homes.get(0).getId();
```

Comment:
What if `homes` is empty?

Reply:
> What if homes is empty?

The Thing goes offline `updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, CONF_ERROR_NO_HOME_ID)` in line 190 below..

Reply:
Hmm no IDE available now, But i would expect `homes.get(0).getId();` to throw an NPE or  out of bounds exception if `homes` is empty

Reply:
Good point! I never tested it because (fortunately) I never had zero homes. => I will add a catch for that.

Reply:
In line 166 we do test `homes.isEmpty()` and return with `updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, CONF_ERROR_NO_HOME)`.

Reply:
ah yes, overlooked that.